### ISSUE TITLE
Re-exported variant is incorrectly marked dead

### ISSUE THREAD

**Issue by sim642** [rescript-association/reanalyze#153]
Running reanalyze https://github.com/rescript-association/reanalyze/commit/389dd682223201b42c33e5c444e5c79e0805adad on Goblint https://github.com/goblint/analyzer/commit/a544002114324a207df72f9e79eb656c3ebe7b5b revealed the following.

In `MyCFG` module we have the following [_re-exporting_](https://v2.ocaml.org/manual/typedecl.html) type definition:
```ocaml
(** Re-exported [Edge.t] with constructors. See [Edge.t] for documentation. *)
type edge = Edge.t =
  | Assign of CilType.Lval.t * CilType.Exp.t
  | Proc of CilType.Lval.t option * CilType.Exp.t * CilType.Exp.t list
  | Entry of CilType.Fundec.t
  | Ret of CilType.Exp.t option * CilType.Fundec.t
  | Test of CilType.Exp.t * bool
  | ASM of string list * Edge.asm_out * Edge.asm_in
  | VDecl of CilType.Varinfo.t
  | Skip
```
and in `Edge` module we have the original type definition:
```ocaml
type t =
  | Assign of CilType.Lval.t * CilType.Exp.t
  (** Assignments lval = exp *)
  | Proc of CilType.Lval.t option * CilType.Exp.t * CilType.Exp.t list
  (** Function calls of the form lva = fexp (e1, e2, ...) *)
  | Entry of CilType.Fundec.t
  (** Entry edge that relates function declaration to function body. You can use
    * this to initialize the local variables. *)
  | Ret of CilType.Exp.t option * CilType.Fundec.t
  (** Return edge is between the return statement, which may optionally contain
    * a return value, and the function. The result of the call is then
    * transferred to the function node! *)
  | Test of CilType.Exp.t * bool
  (** The true-branch or false-branch of a conditional exp *)
  | ASM of string list * asm_out * asm_in
  (** Inline assembly statements, and the annotations for output and input
    * variables. *)
  | VDecl of CilType.Varinfo.t
  (** VDecl edge for the variable in varinfo. Whether such an edge is there for all
    * local variables or only when it is not possible to pull the declaration up, is
    * determined by alwaysGenerateVarDecl in cabs2cil.ml in CIL. One case in which a VDecl
    * is always there is for VLA. If there is a VDecl edge, it is where the declaration originally
    * appeared *)
  | Skip [@dead "t.Skip"] 
  (** This is here for historical reasons. I never use Skip edges! *)
```
Despite the last documentation comment, we use `MyCFG.Skip` in various places and it's not marked dead. But the corresponding `Edge.Skip` variant is marked dead.
This is incorrect, because due to the type re-exporting, it wouldn't be possible to remove `Edge.Skip` as the two type manifests must match for the code to compile. I suppose reanalyze isn't accounting for this fact.

**Comment by cristianoc** [comment]
I think there's no attempt in the implementation to handle re-exported variants at the moment.
